runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/TransformTranslator.java

```diff
@@ -36,7 +36,7 @@
  * A {@link TransformTranslator} knows how to translate a particular subclass of {@link PTransform}
  * for the Cloud Dataflow service. It does so by mutating the {@link TranslationContext}.
  */
-interface TransformTranslator<TransformT extends PTransform> {
+public interface TransformTranslator<TransformT extends PTransform> {
```

Comment:
This is an implementation detail of the `DataflowRunner` that is subject to change at any time. It isn't too crazy to make a configurable translator, but I think that there are a few things we would need to do:

1. Spec out what is configurable and why.
2. Move the composable pieces into utility code, all marked `@Internal` because there is no way we want to commit to backwards compatibility for it.
3. Build the official `DataflowRunner` out of those pieces and completely lock it down, so it is a known supportable piece of software. Then if someone wants to use the same pieces to build a different runner, they can do so and it will not touch the actual `DataflowRunner`